Jenkinsfile

```diff
@@ -0,0 +1,174 @@
+        // See below for what this method does - we're passing an arbitrary environment
+        // variable to it so that JAVA_OPTS and MAVEN_OPTS are set correctly.
+        withMavenEnv(["JAVA_OPTS=-Xmx1536m -Xms512m -XX:MaxPermSize=1024m",
+                      "MAVEN_OPTS=-Xmx1536m -Xms512m -XX:MaxPermSize=1024m"]) {
+            // Actually run Maven!
+            // The -Dmaven.repo.local=${pwd()}/.repository means that Maven will create a
+            // .repository directory at the root of the build (which it gets from the
+            // pwd() Workflow call) and use that for the local Maven repository.
+            sh "mvn -Pdebug -U clean install ${runTests ? '-Dmaven.test.failure.ignore=true -Dconcurrency=1' : '-DskipTests'} -V -B -Dmaven.repo.local=${pwd()}/.repository"
+        }
+    }
+
+    // Once we've built, archive the artifacts and the test results.
+    stage "Archive artifacts and test results"
+
+    step([$class: 'ArtifactArchiver', artifacts: '**/target/*.jar, **/target/*.war, **/target/*.hpi', fingerprint: true])
+    if (runTests) {
+        step([$class: 'JUnitResultArchiver', healthScaleFactor: 20.0, testResults: '**/target/surefire-reports/*.xml'])
+    }
+}
+
+def debFileName
+def rpmFileName
+def suseFileName
+
+// Run the packaging build on a node with the "docker" label.
+timestampedNode('docker') {
+    // First stage here is getting prepped for packaging.
+    stage "packaging - docker prep"
+
+    // Docker environment to build packagings
+    dir('packaging-docker') {
+        git branch: packagingBranch, url: 'https://github.com/jenkinsci/packaging.git'
+        sh 'docker build -t jenkins-packaging-builder:0.1 docker'
+    }
+
+    stage "packaging - actually packaging"
+    // Working packaging code, separate branch with fixes
+    dir('packaging') {
+        deleteDir()
+
+        docker.image("jenkins-packaging-builder:0.1").inside("-u root") {
+            git branch: packagingBranch, url: 'https://github.com/jenkinsci/packaging.git'
+
+            try {
+                // Saw issues with unstashing inside a container, and not sure copy artifact plugin would work here.
+                // So, simple wget.
+                sh "wget -q ${currentBuild.absoluteUrl}/artifact/war/target/jenkins.war"
+                sh "make clean deb rpm suse BRAND=./branding/jenkins.mk BUILDENV=./env/test.mk CREDENTIAL=./credentials/test.mk WAR=jenkins.war"
+            } catch (Exception e) {
+                error "Packaging failed: ${e}"
+            } finally {
+                // Needed to make sure the output of the build can be deleted by later runs.
+                // Hackish, yes, but rpm builds as a numeric UID only user fail, so...
+                sh "chmod -R a+w target || true"
+                sh "chmod a+w jenkins.war || true"
+            }
+            dir("target/debian") {
+                def debFilesFound = findFiles(glob: "*.deb")
+                if (debFilesFound.size() > 0) {
```

Comment:
If the files aren't there, the `sh` step above will fail, so this is more paranoia than anything else.